salt/modules/yumpkg.py

```diff
@@ -1513,18 +1513,20 @@ def install(name=None,
             else:
                 pkgstr = pkgpath
 
-            # Lambda to trim the epoch from the currently-installed version if
-            # no epoch is specified in the specified version
-            norm_epoch = lambda x, y: x.split(':', 1)[-1] \
-                if ':' not in y \
-                else x
             cver = old_as_list.get(pkgname, [])
             if reinstall and cver:
                 for ver in cver:
-                    ver = norm_epoch(ver, version_num)
```

Comment:
Do I understand correctly that this line did not work before this PR in case case `ver` was a dict (`diff_attr == True`)?

Reply:
I just stepped through the code again to double check that. We still need this part, since the version number still includes the epoch if no `diff_attr` is provided. So I was wrong on that. What failed was the splitting of `ver` in `norm_epoch`, since `ver` can be a string and a dict.